### PR TITLE
merge master

### PR DESCRIPTION
**Pull request by rogpeppe**
Merge v5-unstable into new-channels-model.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,7 +38,9 @@ endif
 default: build
 
 $(GOPATH)/bin/godeps:
-	go get -v launchpad.net/godeps
+	# godeps needs to be fetched with the insecure flag as launchpad
+	# uses http for part of the checkout process.
+	go get -v -insecure launchpad.net/godeps
```
Comment (mhilton) [inline Makefile:43]: This should now be github.com/rogpeppe/godeps, but that can wait.

**Comment by mhilton** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1602/
Test PASSed.

**Comment by jrwren** [comment]
:+1:

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**